### PR TITLE
Unit tests are becoming more flaky, so retry them

### PR DESCRIPTION
**Pull request by sagudev**
Especially net test on mac and now also on windows, but I've also seen flaky unit tests on linux.

Hopefully this would make mq faster as there should be less faulty fails.

Temporary solution for https://github.com/servo/servo/issues/30683

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes do not require tests because there are CI resilient changes

### CODE REVIEW

File: .github/workflows/mac.yml [hunk 1]
```diff
@@ -93,8 +93,11 @@ jobs:
         run: ./mach test-scripts
       - name: Unit tests
         if: ${{ inputs.unit-tests || github.ref_name == 'try-mac' }}
-        timeout-minutes: 30 # https://github.com/servo/servo/issues/30275
-        run: python3 ./mach test-unit ${cargo_profile_option}
+        uses: nick-fields/retry@v2
+        with:
+          timeout_minutes: 20 # https://github.com/servo/servo/issues/30275
```
Comment (mrobinson) [inline .github/workflows/mac.yml:98]: Very nice.

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -100,7 +100,10 @@ jobs:
         run: ./mach test-scripts
       - name: Unit tests
         if: ${{ inputs.unit-tests || github.ref_name == 'try-linux' }}
-        run: python3 ./mach test-unit ${cargo_profile_option}
+        uses: nick-fields/retry@v2
+        with:
+          max_attempts: 2
+          command: python ./mach test-unit ${cargo_profile_option}
```
Comment (mrobinson) [inline .github/workflows/linux.yml:106]: I can you put a link to https://github.com/servo/servo/issues/30683 here?

File: .github/workflows/mac.yml [hunk 3]
```diff
@@ -93,8 +93,11 @@ jobs:
         run: ./mach test-scripts
       - name: Unit tests
         if: ${{ inputs.unit-tests || github.ref_name == 'try-mac' }}
-        timeout-minutes: 30 # https://github.com/servo/servo/issues/30275
-        run: python3 ./mach test-unit ${cargo_profile_option}
+        uses: nick-fields/retry@v2
```
Comment (mrobinson) [inline .github/workflows/mac.yml:96]: Ditto.

File: .github/workflows/windows.yml [hunk 4]
```diff
@@ -85,7 +85,10 @@ jobs:
         run: python mach smoketest --angle ($env:cargo_profile_option -split ' ')
       - name: Unit tests
         if: ${{ inputs.unit-tests || github.ref_name == 'try-windows' }}
-        run: python mach test-unit ($env:cargo_profile_option -split ' ')
+        uses: nick-fields/retry@v2
```
Comment (mrobinson) [inline .github/workflows/windows.yml:88]: Please add the link here as well.

**Review by mrobinson — APPROVED** [review]
This needs a rebase, but otherwise this looks good. Let's give this a try!

**Review by mrobinson — APPROVED** [review]

**Comment by sagudev** [comment]
Windows crashed three times, this might depend on which runner we are getting (maybe another runner image update).

**Comment by mrobinson** [comment]
> Windows crashed three times, this might depend on which runner we are getting (maybe another runner image update).

This is so strange. I wonder if the runner is in an unusual state when it starts.

**Comment by sagudev** [comment]
In second and third try unit test are not recompiled so same binaries are run. Maybe we are hitting some nondeterministic compilation problem.

**Comment by mukilan** [comment]
`STATUS_STACK_BUFFER_OVERRUN` [seems to be related](https://stackoverflow.com/questions/76871022/github-build-server-fails-with-status-stack-buffer-overrun) to high memory usage. Out tests don't print 'memory allocation failed' though, so it might be [unrelated](https://devblogs.microsoft.com/oldnewthing/20190108-00/?p=100655)

Is it worth trying to run windows tests sequentially with `--test-thread=1` as suggested?

**Comment by sagudev** [comment]
I am try running `--test-thread=1` in my fork.

**Comment by mrobinson** [comment]
Looking at the try runs on your fork @sagudev, this looks like it's working well. I'm going to try to send this to the merge queue.

**Comment by sagudev** [comment]
Picture of 5 successful Windows CI runs

OFF-TOPIC: Run times are so random.

**Pull request merged**